subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java

```diff
@@ -189,16 +189,22 @@ private void initializeNativeIntegrations(File userHomeDir) {
     }
 
     private void initializeFeatures(EnumSet<NativeFeatures> requestedFeatures) {
-        for (NativeFeatures requestedFeature : requestedFeatures) {
-            if (!initializedFeatures.contains(requestedFeature)) {
-                initializedFeatures.add(requestedFeature);
-                if (requestedFeature.initialize(nativeBaseDir, useNativeIntegrations)) {
-                    enabledFeatures.add(requestedFeature);
+        if (isNativeIntegrationsEnabled()) {
+            for (NativeFeatures requestedFeature : requestedFeatures) {
+                if (!initializedFeatures.contains(requestedFeature)) {
+                    initializedFeatures.add(requestedFeature);
+                    if (requestedFeature.initialize(nativeBaseDir, useNativeIntegrations)) {
+                        enabledFeatures.add(requestedFeature);
+                    }
                 }
```

Comment:
Maybe we could take advantage of the return value of `add`:
```suggestion
                if (initializedFeatures.add(requestedFeature)) {
                    if (requestedFeature.initialize(nativeBaseDir, useNativeIntegrations)) {
                        enabledFeatures.add(requestedFeature);
                    }
                }
```

Reply:
I did that!